python/ray/autoscaler/_private/aws/config.py

```diff
@@ -268,35 +299,39 @@ def _configure_iam_role(config):
         profile.add_role(RoleName=role.name)
         time.sleep(15)  # wait for propagation
 
-    config["head_node"]["IamInstanceProfile"] = {"Arn": profile.arn}
+    head_node_config["IamInstanceProfile"] = {"Arn": profile.arn}
```

Comment:
Ehh...there's some danger here of giving worker nodes too many permissions, in case the head node and some of the workers use the same node type.
The workers of the head type would wind up with permissions to create EC2 instances. 
(These permissions are granted so that the head node can run the autoscaler.)

One approach would be to apply these permissions when launching the head node in `AWSNodeProvider.create_node`. Not sure if it's worth the trouble to deal with this right now, especially given that our example configs all emphasize giving the head node its own type.
Thoughts, author and reviewers @ckw017 @AmeerHajAli @xcharleslin @thomasdesr ?

Reply:
Not necessarily to be done in this PR, but I think the solution is to
(1) Record the head IAM role in the "provider" field.
(2) Have AWSNodeProvider.__init__ store a `self.head_IAM_role`
(3) Have AWSNodeProvider.create_node test `if tags[TAG_RAY_NODE_KIND] == NODE_KIND_HEAD` and insert the `self.head_IAM_role` into the `node_config` before launch.

In retrospect, a separate head node field might have been simpler, oh well.

Reply:
> I quite like @xcharleslin 's idea. I think it is preferable if the configuration is clearly just for the head node (and not for the workers).

Yep, that would work. Then we'd default max_workers to 0 for the head type, possibly remove max_workers for the head type from the example configs, and throw an error if someone tries to set max_workers > 0.

Probably we should figure this another time / place.